### PR TITLE
Add docs for changefeed `execution_locality` option

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-7308, DOC-7337

This PR adds docs for the changefeed `execution_locality` option. Updates include:

- New section to changefeed multi-region deployment page
- Small additions to enterprise process docs
- Option added to create changefeed SQL page
- Mention on cdc queries page in tip
- Added as a feature to the overview page for enterprise changefeeds

## Preview
https://deploy-preview-16878--cockroachdb-docs.netlify.app/docs/v23.1/changefeeds-in-multi-region-deployments.html

### CODE REVIEW

**Review by dt — APPROVED** [review]

**Review by shannonbradshaw — APPROVED** [review]
Just a couple of comments for your consideration.

File: v23.1/change-data-capture-overview.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ The main feature of CDC is the changefeed, which targets an allowlist of tables,
...
+| Streams indefinitely until underlying SQL connection is closed. | Maintains connection to configured sink ([Kafka](changefeed-sinks.html#kafka), [Google Cloud Pub/Sub](changefeed-sinks.html#google-cloud-pub-sub), [Amazon S3](changefeed-sinks.html#amazon-s3), [Google Cloud Storage](changefeed-sinks.html#google-cloud-storage), [Azure Storage](changefeed-sinks.html#azure-blob-storage), [HTTP](changefeed-sinks.html#http), [Webhook](changefeed-sinks.html#webhook-sink)). |
+| Create with [`EXPERIMENTAL CHANGEFEED FOR`](changefeed-for.html). | Create with [`CREATE CHANGEFEED`](create-changefeed.html).<br>Use `CREATE CHANGEFEED` with [CDC queries](cdc-queries.html) to define the emitted change data.<br><span class="version-tag">New in v23.1:</span> Create a scheduled changefeed with [`CREATE SCHEDULE FOR CHANGEFEED`](create-schedule-for-changefeed.html).<br><span class="version-tag">New in v23.1:</span> Use [`execution_locality`](changefeeds-in-multi-region-deployments.html#run-a-changefeed-job-by-locality) to determine node locality for changefeed job execution.  |
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/change-data-capture-overview.md:22]: Yes, they do

File: v23.1/cdc-queries.md [hunk 2]
```diff
@@ -167,6 +167,10 @@ CREATE CHANGEFEED INTO sink AS SELECT * FROM table WHERE crdb_region = 'europe-w
 
 For more detail on targeting `REGIONAL BY ROW` tables with changefeeds, see [Changefeeds in Multi-Region Deployments](changefeeds-in-multi-region-deployments.html).
 
+{{site.data.alerts.callout_success}}
+If you are running changefeeds from a [multi-region](multiregion-overview.html) cluster, you may want to define which nodes take part in running the changefeed job. You can use the [`execution_locality` option](changefeeds-in-multi-region-deployments.html#syntax) with key-value pairs to specify the locality requirements nodes must meet.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/cdc-queries.md]: Done

File: v23.1/change-data-capture-overview.md [hunk 3]
```diff
@@ -37,6 +37,8 @@ With [`resolved`](create-changefeed.html#resolved-option) specified when a chang
 
 As rows are updated, added, and deleted in the targeted table(s), the node sends the row changes through the [rangefeed mechanism](create-and-configure-changefeeds.html#enable-rangefeeds) to the changefeed encoder, which encodes these changes into the [final message format](changefeed-messages.html#responses). The message is emitted from the encoder to the sink—it can emit to any endpoint in the sink. In the diagram example, this means that the messages can emit to any Kafka Broker.
 
+If you are running changefeeds from a [multi-region](multiregion-overview.html) cluster, you may want to define which nodes take part in running the changefeed job. You can use the [`execution_locality` option](changefeeds-in-multi-region-deployments.html#syntax) with key-value pairs to specify the locality requirements nodes must meet. See [Job coordination using the execution locality option](changefeeds-in-multi-region-deployments.html#job-coordination-using-the-execution-locality-option) for detail on how a changefeed job works with this option.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/change-data-capture-overview.md]: Good idea!

File: v23.1/changefeeds-in-multi-region-deployments.md [hunk 4]
```diff
@@ -5,7 +5,57 @@ toc: true
...
+
+{{site.data.alerts.callout_success}}
+To define and filter the change data included in changefeed messages emitted to the sink, see [Change Data Capture Queries](cdc-queries.html).
+{{site.data.alerts.end}}
+
+### Job coordination using the execution locality option
+
+When you start or [resume](resume-job.html) a changefeed with `execution_locality`, it is necessary to determine the coordinating node for the job. If a node that does not match the locality filter is the first node to claim the job, it will find a node that does match the filter and transfer the execution to it. This can result in a short delay in starting or resuming a changefeed job that has execution locality requirements. 
+
+Once the coordinating node is determined, nodes that match the locality requirements will take part in emitting changefeed messages to the sink. The following will happen in different cases:
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/changefeeds-in-multi-region-deployments.md:47]: Yes, and I have added a sentence to that effect. Thanks for pointing out.

**Pull request merged**